your-rents-services-geodata/src/main/java/com/yourrents/services/geodata/repository/ProvinceRepository.java

```diff
@@ -117,6 +121,16 @@ public boolean delete(UUID uuid) {
 				.where(PROVINCE.EXTERNAL_ID.eq(uuid))
 				.fetchOptional(PROVINCE.ID).orElseThrow(
 						() -> new DataNotFoundException("Province not found: " + uuid));
+		Result<Record1<Integer>> result = dsl.select(CITY.ID)
```

Comment:
Not wrong, but a little complicated. What about using dsl.fetchExists instead? (<https://www.jooq.org/javadoc/3.18.x/org.jooq/org/jooq/DSLContext.html#fetchExists(org.jooq.Select)>)

```java
boolean citiesExist = dsl.fetchExists(CITY, CITY.PROVINCE_ID.eq(provinceId));
```

Reply:
ok will change